texar/modules/classifiers/xlnet_classifier.py

```diff
@@ -0,0 +1,173 @@
+                    "type": "Dense",
+                    "kwargs": {
+                        "units": self._encoder.output_size
+                    }
+                })
+
+            # summary type
+            if self._hparams.summary_type == 'last':
+                self.summary_op = lambda output: output[:, -1]
+            elif self._hparams.summary_type == 'first':
+                self.summary_op = lambda output: output[:, 0]
+            elif self._hparams.summary_type == 'mean':
+                self.summary_op = lambda output: tf.reduce_mean(output, axis=1)
+            elif self._hparams.summary_type == 'attn':
+                raise NotImplementedError
+            else:
+                raise ValueError(
+                    f"Unsupported summary type {self._hparams.summary_type}")
+
+            # Creates an dropout layer
+            drop_kwargs = {"rate": self._hparams.dropout}
+            layer_hparams = {"type": "Dropout", "kwargs": drop_kwargs}
+            self._dropout_layer = layers.get_layer(hparams=layer_hparams)
+
+            # Creates an additional classification layer if needed
+            self._num_classes = self._hparams.num_classes
+            if self._num_classes <= 0:
+                self._logit_layer = None
+            else:
+                logit_kwargs = self._hparams.logit_layer_kwargs
+                if logit_kwargs is None:
+                    logit_kwargs = {}
+                elif not isinstance(logit_kwargs, HParams):
+                    raise ValueError(
+                        "hparams['logit_layer_kwargs'] must be a dict.")
+                else:
+                    logit_kwargs = logit_kwargs.todict()
+                logit_kwargs.update({"units": self._num_classes})
+                if 'name' not in logit_kwargs:
+                    logit_kwargs['name'] = "logit_layer"
+
+                layer_hparams = {"type": "Dense", "kwargs": logit_kwargs}
+                self._logit_layer = layers.get_layer(hparams=layer_hparams)
+
+    @staticmethod
+    def default_hparams():
+        hparams = XLNetEncoder.default_hparams()
+        hparams.update({
+            "num_classes": 2,
+            "logit_layer_kwargs": None,
+            "summary_type": "last",
+            "dropout": 0.1,
+            "use_projection": True,
+            "name": "xlnet_classifier"
+        })
+        return hparams
+
+    def _build(self, token_ids, segment_ids=None, input_mask=None, mode=None):
+        if mode is None:
+            mode = global_mode()
+        output, _ = self._encoder(token_ids, segment_ids, input_mask=input_mask,
+                                  mode=mode)
+        summary = self.summary_op(output)
+        if self._hparams.use_projection:
+            summary = tf.tanh(self.projection(summary))
+        # summary: (batch_size, hidden_dim)
+        summary = self._dropout_layer(
+            summary, training=(mode == tf.estimator.ModeKeys.TRAIN))
```

Comment:
`mode` should be a tensor. Here is a good reference of how `mode` is used: https://github.com/asyml/texar/blob/master/texar/modules/embedders/embedders.py#L191

Reply:
Done! [9280366]